Marlin/src/HAL/HAL_STM32/HAL.cpp

```diff
@@ -49,7 +49,7 @@ uint16_t HAL_adc_result;
 // ------------------------
 
 // Needed for DELAY_NS() / DELAY_US() on CORTEX-M7
-#if (defined(__arm__) || defined(__thumb__)) && __CORTEX_M == 7
+#if (defined(__arm__) || defined(__thumb__)) && WITHIN(__CORTEX_M, 3, 7)
```

Comment:
WITHIN is defined as :
#define WITHIN(N,L,H)       ((N) >= (L) && (N) <= (H))

Obviously not what you where going for there?

@thinkyhead   I don't know if you have another macro to say if __CORTEX_M IN (3,7) which i guess is what was being tried here?